### ISSUE TITLE
BUG: shared log axes lose _minpos and revert to default

### ISSUE THREAD

**Issue by QuLogic** [matplotlib/matplotlib#7587]
This example showcases two instances where the view limits in 1.5.3 are not the same as when using 2.0.0rc1 with the classic style.

```python
from distutils.version import LooseVersion

import numpy as np
import matplotlib as mpl
import matplotlib.pyplot as plt


if LooseVersion(mpl.__version__) >= LooseVersion('2.0.0'):
    plt.style.use('classic')


cc_t = np.arange(-100, 101, 5) / 1000
cc_concave = np.ones_like(cc_t)

fig, ax = plt.subplots()
ax.plot(cc_t, cc_concave, ls="", marker=".", color="0.7",
        label="xcorr (concave)")
ax.set_title(mpl.__version__)


np.random.seed(0)
freq = np.arange(0.0, 100.0, 0.001)
resp = np.random.rand(*freq.shape) + 1j * np.random.rand(*freq.shape)

fig, (ax1, ax2) = plt.subplots(2, 1, sharex=True)
ax1.loglog(freq, abs(resp), lw=1.5)
ax1.set_title(mpl.__version__)
ax2.semilogx(freq, np.angle(resp), lw=1.5)
ax2.set_title(mpl.__version__)

plt.show()
```

For the first plot, with 1.5.3, the resulting image, with view limits (-0.1, 0.15), is:

![figure_1](https://cloud.githubusercontent.com/assets/302469/20993916/4a644628-bcbb-11e6-87de-5650b6a901c9.png)

but with 2.0.0rc1 in classic style, its view limits are (-0.1, 0.1):
![figure_1-2 0 0rc1](https://cloud.githubusercontent.com/assets/302469/20993918/4e34d876-bcbb-11e6-9ea5-59afded7cb4f.png)

For the second plot, with 1.5.3, the result image, with view limits (10^-3, 10^2), is:
![figure_2](https://cloud.githubusercontent.com/assets/302469/20993988/af7f4aa8-bcbb-11e6-8b02-cdbc417bf1ca.png)

but with 2.0.0rc1 in classic style, its view limits are (10^-7, 10^2):
![figure_2-2 0 0rc1](https://cloud.githubusercontent.com/assets/302469/20993991/b37678a2-bcbb-11e6-9e3a-de597b787795.png)

The interesting thing is that this view limit is only triggered if something is plotted in both axes.

**Comment by efiring** [comment]
The first example is the sort of thing we have to expect; classic style brings things close to 1.5.3, but trying to guarantee identical results is the road to insanity.  There is nothing wrong with the 2.0 classic version.
The second example is pointing to a major bug in how the invalid zero frequency value is handled with shared axes.  The problem is that somehow, having the second log axes is causing the default `Bbox._minpos` value of 1e-7 to override the actual minimum positive value.

**Comment by efiring** [comment]
That second example looks familiar... It seems like we have dealt with this, or at least had it pointed out, some time ago.  Or maybe I saw it in devdocs but passed over it without investigating.
The bug is not specific to classic style; it also occurs with the 2.0 default style.

**Comment by QuLogic** [comment]
Maybe you were thinking of #7410? It seems to be a similar topic.

**Comment by efiring** [comment]
This is related to #7493.

**Comment by QuLogic** [comment]
Closed by #7598.
